File: python/p5.py
import math

def lcm(a, b):
	return a * b // math.gcd(a, b)

def smallest_multiple(numbers):
	lcm_of_numbers = numbers[0]

	for number in numbers[1:]:
		lcm_of_numbers = lcm(lcm_of_numbers, number)
	return lcm_of_numbers


if __name__ == "__main__":
	numbers = range(1, 21)
	smallest_multiple = smallest_multiple(numbers)
	print(smallest_multiple)

File: go/p5.go
package main

func gcd(a, b int) int {
	for b != 0 {
		t := b
		b = a % b
		a = t
	}
	return a
}

func lcm(a, b int) int {
	return a * b / gcd(a, b)
}

func p5() int {
	smallestMultiple := 1
	for i := 2; i <= 20; i++ {
		smallestMultiple = lcm(smallestMultiple, i)
	}
	return smallestMultiple
}
